zeroconf/__init__.py

```diff
@@ -1765,12 +1795,13 @@ def __init__(
         other_ttl: int = _DNS_OTHER_TTL,
         *,
         addresses: Optional[List[bytes]] = None,
-        parsed_addresses: Optional[List[str]] = None
+        parsed_addresses: Optional[List[str]] = None,
+        strict: bool = True
     ) -> None:
         # Accept both none, or one, but not both.
         if addresses is not None and parsed_addresses is not None:
             raise TypeError("addresses and parsed_addresses cannot be provided together")
-        if not type_.endswith(service_type_name(name, allow_underscores=True)):
+        if not type_.endswith(service_type_name(name, strict=strict, allow_underscores=True)):
```

Comment:
I'm of an opinion that this is not the place to determine whether we're strict or not – `ServiceInfo` is constructed in both receiving and sending context. Let's do it the same way `allow_underscores` is used currently, so:
* pass `strict=False` here
* make `check_service` for sending use strict mode when calling `service_type_name`

Reply:
PS. The same way I'd not expose `strict` in `ServiceInfo`'s constructor

Reply:
Will make `strict=False`. `check_service` should be good since the default behavior is `strict=True`

Will remove `strict` from `ServiceInfo`'s constructor